automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/apps/AppCreationSpec.scala

```diff
@@ -16,7 +16,8 @@ import scala.concurrent.duration._
 
 @DoNotDiscover
 class AppCreationSpec extends GPAllocFixtureSpec with LeonardoTestUtils with GPAllocUtils with ParallelTestExecution {
-  implicit val auth: Authorization =
+  // Using def instead of val to prevent test flakiness due to token expiration when tests take long
+  implicit def auth: Authorization =
     Authorization(Credentials.Token(AuthScheme.Bearer, ronCreds.makeAuthToken().value))
```

Comment:
`Credentials` in `workbench-libs` caches tokens for 60 mins so I don't expect a performance penalty per se.

Reply:
Whats the Mechanism by which this would be refreshed? I thought specs were more or less singletons created at the time of test suite launch. Maybe make it a lazy val to defer its initialization (the 60min countdown) to the point where it is used?

Reply:
I'm a bit less concerned about performance penalty but more introducing flakiness in a different way since getting auth token is a good call as well...maybe just add a line for grabbing new token before the line where token is likely to expire?

making it lazy val might not be sufficient if the test is taking too long....

Reply:
I think `def` is correct given there's already a [60m cache in wb-libs](https://github.com/broadinstitute/workbench-libs/blob/develop/serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/config/Credentials.scala#L16). This method is called for every call to `LeonardoApiClient`. It's better I think to have a guaranteed fresh token and not add any additional caching to Leo via `val` or `lazy val`.

Reply:
oh...in that case, definitely makes sense to have `def`

Reply:
> Whats the Mechanism by which this would be refreshed? I thought specs were more or less singletons created at the time of test suite launch (which would mean that is when this token is initialized as well).

The API call methods defined in `LeonardoApiClient` like [this](https://github.com/DataBiosphere/leonardo/blob/f2bea48050dc0a391b8bc0741ce564f9500ad630/automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoApiClient.scala#L541) takes an `authHeader: Authorization` instance as an implicit input. Changing `val` into `def` should force `ronCreds.makeAuthToken()` to be re-evaluated at every API method call. Re-evaluation means, if a cached token has been generated less than 60 mins ago, it'll be re-used. After 60 mins, cache should be invalidated and a new token should be generated. This should fit into our use case nicely.